sympy/solvers/solvers.py

```diff
@@ -964,6 +964,7 @@ def _sympified_list(w):
 
         # if we have a Matrix, we need to iterate over its elements again
         if f[i].is_Matrix:
+            f[i] = f[i].as_explicit()
```

Comment:
What is needed now is to catch the exception here and then raise a new exception with a better error message saying that `solve` cannot handle matrices with symbolic shape.